Rework poetry dependencies and group.

- Remove `editorconfig-checker` and `poetry-lock-package` dependencies.
- Add group `lint` containing `ruff`, `black` and `mypy` dependencies.
- Add group `packaging`.
- Move `async_exit_stack` to `test` group.
# ------------------------ >7 ------------------------
# poetry.lock    | 30 +++---------------------------
# pyproject.toml | 31 +++++++++++++++++++++----------
# 2 files changed, 24 insertions(+), 37 deletions(-)
diff --git a/poetry.lock b/poetry.lock
index 250cd2ee8..036ba76c9 100644
--- a/poetry.lock
+++ b/poetry.lock
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.6.1 and should not be changed by hand.
+# This file is automatically @generated by Poetry 1.5.1 and should not be changed by hand.

 [[package]]
 name = "aiofiles"
@@ -957,16 +957,6 @@ files = [
 testing = ["mock", "pytest", "pytest-cov"]
 webassets = ["webassets"]

-[[package]]
-name = "editorconfig-checker"
-version = "2.4.0"
-description = "Python wrapper around invoking editorconfig-checker (https://github.com/editorconfig-checker/editorconfig-checker)"
-optional = false
-python-versions = ">=2.7"
-files = [
-    {file = "editorconfig-checker-2.4.0.tar.gz", hash = "sha256:1f7839df7d967d95f0cf2ed6a42a0f456c65072d2ffcff1fedc0eac62ed236ea"},
-]
-
 [[package]]
 name = "exceptiongroup"
 version = "1.0.4"
@@ -1570,20 +1560,6 @@ files = [
 dev = ["pre-commit", "tox"]
 testing = ["pytest", "pytest-benchmark"]

-[[package]]
-name = "poetry-lock-package"
-version = "0.4.5"
-description = "Poetry lock package generator"
-optional = false
-python-versions = ">=3.6.2,<4.0.0"
-files = [
-    {file = "poetry-lock-package-0.4.5.tar.gz", hash = "sha256:a74a4916cc53ebbfd3b3575b73d0e93bac74b75ff2d74f92a24e7a8938fc20f2"},
-    {file = "poetry_lock_package-0.4.5-py3-none-any.whl", hash = "sha256:5b8ae966ec8c760a69d38bde6887501d594b467e00902dd341328cd8079ccaee"},
-]
-
-[package.dependencies]
-toml = ">=0.10.1,<0.11.0"
-
 [[package]]
 name = "priority"
 version = "2.0.0"
@@ -2551,7 +2527,7 @@ reco = ["python-dateutil", "simplejson"]
 name = "toml"
 version = "0.10.2"
 description = "Python Library for Tom's Obvious, Minimal Language"
-optional = false
+optional = true
 python-versions = ">=2.6, !=3.0.*, !=3.1.*, !=3.2.*"
 files = [
     {file = "toml-0.10.2-py2.py3-none-any.whl", hash = "sha256:806143ae5bfb6a3c6e736a764057db0e6a0e05e338b5630894a5f779cabb4f9b"},
@@ -2827,4 +2803,4 @@ parsec-ext-deps = ["asn1crypto", "certvalidator", "oscrypto"]
 [metadata]
 lock-version = "2.0"
 python-versions = "~3.9.10"
-content-hash = "fca980702e679331fdf55173e529633cf54b8a793e2642b8db0aa1538f47e7d7"
+content-hash = "c2b99f1d750ee3be85ffbc8e85e227e554b98264d7ed514d98bf4b09e4db6531"
diff --git a/pyproject.toml b/pyproject.toml
index 0accef9c6..7317f856f 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -55,7 +55,6 @@ trio_typing = "^0.7"
 sentry-sdk = ">=1.13.0,<1.26.0"
 structlog = ">=21.5,<23.0"
 # colorama = "^0.4.4  # structlog colored output"
-async_exit_stack = "^1.0"
 outcome = "^1.1"

 # Backend-only requirements
@@ -128,7 +127,9 @@ parsec-ext-deps = [
     "certvalidator",
 ]

-[tool.poetry.group.dev.dependencies]
+[tool.poetry.group.test]
+
+[tool.poetry.group.test.dependencies]
 pytest = "^7.0"
 pytest-cov = "^4.0"
 pytest-xdist = "^3.1"
@@ -137,31 +138,41 @@ pytest-qt = "^4.0"
 pytest-rerunfailures = "^10.2"
 hypothesis = "^6.54"
 hypothesis-trio = "^0.6"
+async_exit_stack = "^1.0"
+
+[tool.poetry.group.dev.dependencies]
 trustme = "^0.9"
 winfspy = { version="^0.8", extras=["test"], markers="platform_system=='Windows'" }
 psutil = "^5.9"
-poetry-lock-package = "^0.4.4"
 docutils = "0.18.1"
 Babel = "^2.10"
 PyQt5_stubs = "^5.15"
-# PyInstaller is only needed for packaging, but it must be installed in the same
-# venv the code to package is installed. Hence it's simpler to have it here so
-# it dependencies won't clash with the rest of the project.
-pyinstaller = "^5.3"
+boto3-stubs = "^1.26"
+types-requests = "^2.28"

+[tool.poetry.group.lint]
+optional = true
+
+[tool.poetry.group.lint.dependencies]
 ruff = "0.1.14"
 black = "23.3.0"
 mypy = "^0.982"

+[tool.poetry.group.packaging]
+optional = true
+
+[tool.poetry.group.packaging.dependencies]
+# PyInstaller is only needed for packaging, but it must be installed in the same
+# venv the code to package is installed. Hence it's simpler to have it here so
+# it dependencies won't clash with the rest of the project.
+pyinstaller = { version = "^5.3", markers = "platform_system!='Linux'" }
+
 setuptools = ">=63.1,<68.0"

 maturin = "0.15.1"
 patchelf = { version = "^0.17.2.0", markers = "platform_system=='Linux'" }

 cibuildwheel = "v2.16.5"
-editorconfig-checker = "2.4.0"
-types-requests = "^2.28"
-boto3-stubs = "^1.26"

 [tool.poetry.group.docs]
 optional = true
--
2.42.0
